homeassistant/components/amcrest/camera.py

```diff
@@ -170,18 +208,183 @@ def turn_on(self):
         """Turn on camera."""
         self._enable_video_stream(True)
 
-    # Utility methods
+    def enable_motion_detection(self):
+        """Enable motion detection in the camera."""
+        self._enable_motion_detection(True)
+
+    def disable_motion_detection(self):
+        """Disable motion detection in camera."""
+        self._enable_motion_detection(False)
+
+    # Additional Amcrest Camera service methods
+
+    @callback
+    def async_enable_recording(self):
+        """Call the job and enable recording."""
+        return self.hass.async_add_job(self._enable_recording, True)
```

Comment:
Use `self.hass.async_add_executor_job` instead.

Reply:
I assume you mean for all instances. Will do.